### PR TITLE
virt-api: Remove the deprecated IsKubeVirtServiceAccount function

### LINKED ISSUE #11177: Optimize IsKubeVirtServiceAccount()
**Is your feature request related to a problem? Please describe**:
https://github.com/kubevirt/kubevirt/blob/f081b2bf56f629788e46359c8d2e008fbe7834a7/pkg/virt-api/webhooks/utils.go#L83
`IsKubeVirtServiceAccount` is called from:
- VMI mutating webhook
- VMI creation validating webhook
- VMI update validating webhook

Every time this function is called, it [reads](https://github.com/kubevirt/kubevirt/blob/17517e765488f3dfd0c4272781d56522dc6051f4/staging/src/kubevirt.io/client-go/util/util.go#L13) a static file containing the namespace in which the `virt-api` pod is running.

**Describe the solution you'd like**:
Refactor the code in a way that will read the namespace file only once.

**Describe alternatives you've considered**:

**Additional context**:

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
Currently, the deprecated `IsKubeVirtServiceAccount` is only called from `ValidateVirtualMachineInstanceMetadata`.

Adjust its signature to take in a boolean to indicate whether the request had originated from a KubeVirt ServiceAccount or not.

Recursively adjust the signatures of all its callers to match.

Fixes #11177.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go [hunk 1]
```diff
@@ -41,7 +41,8 @@ import (
 )
 
 type VMPoolAdmitter struct {
-	ClusterConfig *virtconfig.ClusterConfig
+	ClusterConfig           *virtconfig.ClusterConfig
+	KubeVirtServiceAccounts map[string]struct{}
```
Comment (iholder101) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go:45]: Can you please unexpose this field? Same goes for all the similar fields

Comment (iholder101) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go:45]: Why `map[string]struct{}` and not `map[string]string`?

File: pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go [hunk 2]
```diff
@@ -137,7 +138,9 @@ func (admitter *VMsAdmitter) Admit(ctx context.Context, ar *admissionv1.Admissio
 			return webhookutils.ToAdmissionResponse(causes)
 		}
 	}
-	causes = ValidateVirtualMachineSpec(k8sfield.NewPath("spec"), &vmCopy.Spec, admitter.ClusterConfig, accountName)
+
+	_, isKubeVirtServiceAccount := admitter.KubeVirtServiceAccounts[ar.Request.UserInfo.Username]
```
Comment (iholder101) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go:142]: nit: WDYT about something like
```go
func isArOfKubevirtServiceAccount(ar *AdmissionReview, KubeVirtServiceAccounts map[string]struct{}) bool {}
```

Although I'm not sure it's more elegant than the current code.. WDYT?

**Review by iholder101 — COMMENTED** [review]
Thanks @orelmisan! Left a few small comments below

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go (hunk 1 above)
Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go:45]: Unfortunately, just a few admitters have a constructor function.
Even more unfortunate is the fact that `VMsAdmitter` has a constructor function, but is also initialized without it.
I will create a dedicated PR to fix it.
Having it here will increase the scope of this PR.

Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmpool-admitter.go:45]: The map contain keys which are KubeVirt's ServiceAccount full names (including the namespace KubeVirt is deployed to).
We only care if the map contains the key - thus no need to have a value.

**Comment by iholder101** [comment]
Thanks @orelmisan!

/approve

File: pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go (hunk 2 above)
Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go:142]: TBH, I don't see the benefit of adding this function at the moment.
The admitters will still have to contain the KubeVirt ServiceAccount map / set, and the complex validation logic mostly care if the change was made by KubeVirt or not.

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm
Nice PR and division of commits!

**Pull request merged**